src/coins.cpp

```diff
@@ -1460,16 +1516,22 @@ CValidationState::Code CCoinsViewCache::IsScTxApplicableToState(const CTransacti
                 __func__, __LINE__, tx.ToString(), csw.ToString());
             return CValidationState::Code::INVALID;
         }
-        
-        int32_t idx = csw.actCertDataIdx;
-        // index should have been already checked at this point, at() throws an exception
-        CFieldElement certDataHash = tx.GetVActCertData().at(idx);
-        if (GetActiveCertDataHash(csw.scId) != certDataHash)
+
+        CScCertificateView certView = this->GetActiveCertView(csw.scId);
+        // note: it's also fine to have actCertDataHash == CFieldElement()
```

Comment:
regarding the note: both certView.certDataHash, csw.actCertDataHash have to be == CFieldElement() to be valid

Reply:
Done